client/notifications/src/panel/state/index.js

```diff
@@ -16,13 +22,13 @@ const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
 const withMiddleware = customMiddleware =>
 	composeEnhancers( applyMiddleware( actionMiddleware( customMiddleware ) ) )( createStore );
 
-export const init = ( { customEnhancer, customMiddleware = {} } = {} ) => {
+export let store = init();
+
+export function init( { customEnhancer, customMiddleware = {} } = {} ) {
 	const middle = withMiddleware( customMiddleware );
 	const create = customEnhancer ? customEnhancer( middle ) : middle;
 
 	store = create( reducer, reducer( undefined, { type: '@@INIT' } ) );
 
 	return store;
-};
```

Comment:
is it not a bit funny hat we're using `store = init()` when `init()` not only _returns_ `store` but also sets it directly?

I would have expected maybe just returning a new value for store but not setting it through the function-call…

```js
export let store = init();

export function init( … ) {
	const store = create( … );
	return store;
}
```

this is an interesting change and I think I ran into a similar problem in some experimental code I was writing for this a couple years ago. at the time I rewrote `store` (which in my case was `wpcom`) as a function to get around it, but I think I like this approach better, though it does seem more magical

```js
let appStore;

export const store = () => {
	if ( ! appStore ) {
		appStore = init();
	}

	return appStore;
}
```